core/domain/topic_jobs_one_off.py

```diff
@@ -92,3 +92,55 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class UpdateTopicThumbnailSizeOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to update the thumbnail_size_in_bytes in topic models. """
+
+    _DELETED_KEY = 'topic_deleted'
+    _ERROR_KEY = 'update_error'
+    _SUCCESS_KEY = 'thumbnail_size_updated'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [topic_models.TopicModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        topic = topic_fetchers.get_topic_by_id(item.id)
+
+        try:
+            # We are not updating thumbnail_filename here, but using it call the
+            # update for topic thumbnail_size_in_bytes.
+            # old_value and new_value are same here, because the update for
+            # thumbnail_size_in_bytes is called from within the code for
+            # updating thumbnail_filename in topic_services.py file.
+            commit_cmds = [topic_domain.TopicChange({
```

Comment:
I'm not really sure why you need to do all this. Couldn't you just operate directly on the TopicModel? Extract the image field, find its size, and do a put() (without changing the last-updated time) -- you don't need to increment the version (which would happen with the topic_services methods) since there is no user-initiated change.

Reply:
Agreed, after looking up other examples, I learnt that we don't need to commit a change. We can instead work with the models directly.